ruby/ql/lib/codeql/ruby/dataflow/internal/DataFlowPrivate.qll

```diff
@@ -378,12 +378,16 @@ private module Cached {
   }
 
   cached
-  newtype TContentSet =
+  newtype TOptionalContentSet =
     TSingletonContent(Content c) or
     TAnyElementContent() or
     TElementLowerBoundContent(int lower) {
       FlowSummaryImplSpecific::ParsePositions::isParsedElementLowerBoundPosition(_, lower)
-    }
+    } or
+    TNoContentSet()
```

Comment:
Couldn't this rather be defined as a `newtype` inside `TypeTracker.qll`:
```ql
newtype TOptionalContentSet =
  TContentSetSome(ContentSet cs) or
  TContentSetNone()
```

Reply:
You're asking if I want to revert https://github.com/github/codeql/pull/10375/commits/b1c18986483520926ea004c81d7b8b7bebbef983?

Repeatedly boxing values is just less efficient than putting them in the same universe. The `Optional` wrapper is the most obvious one to avoid.